### ISSUE TITLE
infer.py与predict_cls.py推理的结果不一致

### ISSUE THREAD

**Issue by 176050800** [PaddlePaddle/PaddleClas#2077]
在Ubuntu18.04/Python3.7/CUDA11.1下，用paddleclas做图像分类测试的时候，tools/infer.py与deploy/python/predict_cls.py推理的结果完全不一致，图像一共分三类(very_good/good/no_good)，infer.py推理的结果是OK的，但predict_cls.py推理的结果偏差很大，是export_model.py在导出inference模型的时候出了问题吗？
执行tools/infer.py：
python3 tools/infer.py -c ppcls/configs/face_state/PPLCNet_x1_0.yaml -o Global.pretrained_model=output/PPLCNet_x1_0/latest -o Infer.infer_imgs=dataset/face_state/train/very_good/003.jpg
预测结果：
[{'class_ids': [0], 'scores': [0.99498], 'file_name': 'dataset/face_state/train/very_good/003.jpg', 'label_names': ['very_good']}]

再执行tools/export_model.py，生成inference模型：
python3 tools/export_model.py -c ppcls/configs/face_state/PPLCNet_x1_0.yaml -o Global.pretrained_model=output/PPLCNet_x1_0/latest

执行deploy/python/predict_cls.py：
python3 deploy/python/predict_cls.py -c deploy/configs/face_state/inference.yaml
预测结果：
003.jpg:	class id(s): [1, 2], score(s): [0.43, 0.38], label_name(s): ['good', 'no_good']

两个yaml文件的参数设置是一致的。

**Comment by cuicheng01** [comment]
流程上没有问题。
1.`deploy/configs/face_state/inference.yaml`确认下这个里边的图片是和infer.py预测的是一张图片吗？
2.确认下`deploy/configs/face_state/inference.yaml`中指定的inference model是导出的模型的路径吗？

**Comment by 176050800** [comment]
> 流程上没有问题。 1.`deploy/configs/face_state/inference.yaml`确认下这个里边的图片是和infer.py预测的是一张图片吗？ 2.确认下`deploy/configs/face_state/inference.yaml`中指定的inference model是导出的模型的路径吗？

预测的图片是同一张，该图片属于训练集中的其中之一；inference model也确认过没有问题，属于export指定的路径，下面为两个yaml的详细参数：

ppcls/configs/face_state/PPLCNet_x1_0.yaml：
# global configs
Global:
  checkpoints: null
  pretrained_model: null
  output_dir: ./output/
  device: gpu
  save_interval: 10
  eval_during_train: True
  eval_interval: 10
  epochs: 40
  print_batch_step: 10
  use_visualdl: False
  # used for static mode and model export
  image_shape: [3, 224, 224]
  save_inference_dir: ./inference

# model architecture
Arch:
  name: PPLCNet_x1_0
  pretrained: True
  use_ssld: True
  class_num: 3
  use_sync_bn : True
 
# loss function config for traing/eval process
Loss:
  Train:
    - CELoss:
        weight: 1.0
        epsilon: 0.1
  Eval:
    - CELoss:
        weight: 1.0

Optimizer:
  name: Momentum
  momentum: 0.9
  lr:
    name: Cosine
    learning_rate: 0.025
    warmup_epoch: 5
  regularizer:
    name: 'L2'
    coeff: 0.00003


# data loader for train and eval
DataLoader:
  Train:
    dataset:
      name: ImageNetDataset
      image_root: ./dataset/face_state/
      cls_label_path: ./dataset/face_state/train_list.txt
      transform_ops:
        - DecodeImage:
            to_rgb: True
            channel_first: False
        - RandCropImage:
            size: 176
        - RandFlipImage:
            flip_code: 1
        - TimmAutoAugment:
            prob : 0.5
            config_str: rand-m9-mstd0.5-inc1
            interpolation: bicubic
            img_size : 176
        - NormalizeImage:
            scale: 1.0/255.0
            mean: [0.485, 0.456, 0.406]
            std: [0.229, 0.224, 0.225]
            order: ''
        - RandomErasing:
            EPSILON : 0.1
            r1 : 0.3
            sh : 1.0/3.0
            sl : 0.02
            attempt : 10
            use_log_aspect : True
            mode : pixel
    sampler:
      name: DistributedBatchSampler
      batch_size: 64
      drop_last: False
      shuffle: True
    loader:
      num_workers: 8
      use_shared_memory: True

  Eval:
    dataset: 
      name: ImageNetDataset
      image_root: ./dataset/face_state/
      cls_label_path: ./dataset/face_state/val_list.txt
      transform_ops:
        - DecodeImage:
            to_rgb: True
            channel_first: False
        - ResizeImage:
            resize_short: 256
        - CropImage:
            size: 224
        - NormalizeImage:
            scale: 1.0/255.0
            mean: [0.485, 0.456, 0.406]
            std: [0.229, 0.224, 0.225]
            order: ''
    sampler:
      name: DistributedBatchSampler
      batch_size: 64
      drop_last: False
      shuffle: False
    loader:
      num_workers: 4
      use_shared_memory: True

Infer:
  infer_imgs: ./dataset/face_state/train/very_good/003.jpg
  batch_size: 1
  transforms:
    - DecodeImage:
        to_rgb: True
        channel_first: False
    - ResizeImage:
        resize_short: 256
    - CropImage:
        size: 224
    - NormalizeImage:
        scale: 1.0/255.0
        mean: [0.485, 0.456, 0.406]
        std: [0.229, 0.224, 0.225]
        order: ''
    - ToCHWImage:
  PostProcess:
    name: Topk
    topk: 1
    class_id_map_file: dataset/face_state/class.txt

Metric:
  Train:
    - TopkAcc:
        topk: [1]
  Eval:
    - TprAtFpr:
        max_fpr: 0.0001
    - TopkAcc:
        topk: [1]

deploy/configs/face_state/inference.yaml：
Global:
  infer_imgs: "./dataset/face_state/train/very_good/003.jpg"
  inference_model_dir: "./inference/"
  batch_size: 1
  use_gpu: True
  enable_mkldnn: False
  cpu_num_threads: 10
  enable_benchmark: True
  use_fp16: False
  ir_optim: True
  use_tensorrt: False
  gpu_mem: 8000
  enable_profile: False

PreProcess:
  transform_ops:
    - ResizeImage:
        resize_short: 256
    - CropImage:
        size: 224
    - NormalizeImage:
        scale: 0.00392157
        mean: [0.485, 0.456, 0.406]
        std: [0.229, 0.224, 0.225]
        order: ''
        channel_num: 3
    - ToCHWImage:

PostProcess:
  main_indicator: Topk
  Topk:
    topk: 2
    class_id_map_file: ./dataset/face_state/class.txt
  SavePreLabel:
    save_dir: ./pre_label/

**Comment by cuicheng01** [comment]
注意到两个yaml中的`inference`路径应该不一样，在`deploy/configs/face_state/inference.yaml`中的yaml指定的`inference`应该是`../inference` 吧？

**Comment by 176050800** [comment]
应该是我的训练集图片过少导致的该问题，利用官方提供的safety_helmet训练集进行测试，没有该问题，然后我将自己的数据集由原先的100+张图片提升到2000+多张，再测试，没有发现该问题了。
